auth/src/test/java/com/firebase/ui/auth/testhelpers/TestHelper.java

```diff
@@ -15,44 +15,120 @@
+        AuthUI.setApplicationContext(RuntimeEnvironment.application);
+        FirebaseApp app = initializeApp(RuntimeEnvironment.application);
+        injectMockFirebaseAuth(app);
+    }
+
+    private static FirebaseApp initializeApp(Context context) {
         try {
             return FirebaseApp.initializeApp(
                     context,
                     new FirebaseOptions.Builder()
                             .setApiKey(API_KEY)
                             .setApplicationId(APPLICATION_ID)
                             .build(),
-                    FIREBASE_APP_NAME);
+                    FirebaseApp.DEFAULT_APP_NAME);
         } catch (IllegalStateException e) {
-            return FirebaseApp.getInstance(FIREBASE_APP_NAME);
+            return FirebaseApp.getInstance(FirebaseApp.DEFAULT_APP_NAME);
+        }
+    }
+
+    /**
+     * This method finds the map of FirebaseAuth instances and injects of a mock instance associated
+     * with the given FirebaseApp for testing purposes.
+     */
+    private static void injectMockFirebaseAuth(FirebaseApp app) {
+        for (Field field : FirebaseAuth.class.getDeclaredFields()) {
+            field.setAccessible(true);
+
+            Object o;
+            try {
+                o = field.get(null);
+            } catch (IllegalAccessException e) {
+                throw new IllegalStateException(e);
+            } catch (NullPointerException e) {
+                continue;
+            }
+
+            Type genericType = field.getGenericType();
+            if (o instanceof Map && genericType instanceof ParameterizedType) {
+                Type[] parameterTypes = ((ParameterizedType) genericType).getActualTypeArguments();
+                if (parameterTypes.length != 2 || parameterTypes[0] != String.class
+                        || parameterTypes[1] != FirebaseAuth.class) {
+                    continue;
+                }
+
+                //noinspection unchecked
+                Map<String, FirebaseAuth> instances = (Map<String, FirebaseAuth>) o;
+
+                FirebaseAuth.getInstance(app);
+                for (String id : instances.keySet()) {
+                    instances.put(id, mock(FirebaseAuth.class));
+                }
+
+                break;
+            }
         }
+
+        when(FirebaseAuth.getInstance(app).setFirebaseUIVersion(anyString()))
+                .thenReturn(Tasks.<Void>forResult(null));
+    }
+
+    private static void initializeProviders() {
+        RuntimeEnvironment.application = spy(RuntimeEnvironment.application);
+        when(RuntimeEnvironment.application.getApplicationContext())
+                .thenReturn(RuntimeEnvironment.application);
+        Resources spiedResources = spy(RuntimeEnvironment.application.getResources());
+        when(RuntimeEnvironment.application.getResources()).thenReturn(spiedResources);
+
+        when(spiedResources.getString(R.string.default_web_client_id)).thenReturn("abc");
```

Comment:
nit: we might want to extract this into a test helper that:

  * Always spies the runtime application context
  * Provides a static method for resource override

Seems generally useful

Reply:
SGTM!